cleanlab/datalab/datalab.py

```diff
@@ -0,0 +1,632 @@
+        )
+
+        issue_types_copy = self._set_issue_types(issue_types, required_args_per_issue_type)
+
+        new_issue_managers = [
+            factory(datalab=self, **issue_types_copy.get(factory.issue_name, {}))
+            for factory in _IssueManagerFactory.from_list(list(issue_types_copy.keys()))
+        ]
+
+        failed_managers = []
+        for issue_manager, arg_dict in zip(new_issue_managers, issue_types_copy.values()):
+            try:
+                if self.verbosity:
+                    print(f"Finding {issue_manager.issue_name} issues ...")
+                issue_manager.find_issues(**arg_dict)
+                self.data_issues.collect_statistics_from_issue_manager(issue_manager)
+                self.data_issues._collect_results_from_issue_manager(issue_manager)
+            except Exception as e:
+                print(f"Error in {issue_manager.issue_name}: {e}")
+                failed_managers.append(issue_manager)
+
+        if self.verbosity:
+            print("Search complete.")
+        if failed_managers:
+            print(f"Failed to find issues for {failed_managers}")
+
+        self.data_issues.set_health_score()
+
+    def get_info(self, issue_name: Optional[str] = None) -> Dict[str, Any]:
+        """Returns dict of info about a specific issue,
+        or None if this issue does not exist in self.info.
+        Internally fetched from self.info[issue_name] and prettified.
+        Keys might include: number of examples suffering from issue,
+        indicates of top-K examples most severely suffering,
+        other misc stuff like which sets of examples are duplicates if the issue=="duplicated".
+        """  # TODO: Revise Datalab.get_info docstring
+        return self.data_issues.get_info(issue_name)
+
+    def report(
+        self,
+        num_examples: int = 5,
+        verbosity: Optional[int] = None,
+        include_description: bool = True,
+    ) -> None:
+        """Prints informative summary of all issues.
+
+        Parameters
+        ----------
+        num_examples :
+            Number of examples to show for each type of issue.
+            The report shows the top `num_examples` instances in the dataset that suffer the most from each type of issue.
+
+        verbosity :
+            Higher verbosity levels add more information to the report.
+        
+        include_description :
+            Whether or not to include a description of each issue type in the report.
+            Consider setting this to ``False`` once you're familiar with how each issue type is defined.  
+        """
+        if verbosity is None:
+            verbosity = self.verbosity
+        # Show summary of issues
+        print(
+            self._get_report(
+                num_examples=num_examples,
+                verbosity=verbosity,
+                include_description=include_description,
+            )
+        )
+
+    def save(self, path: str) -> None:
```

Comment:
TODO: make this safer. There should be an `overwrite` (or similar named) arg with default value = False. 
Only if its True do you allow this method to overwrite an existing folder.

Look at how cleanvision save/load does this safely and just copy it here.

Reply:
Added:

On the Datalab side:
https://github.com/cleanlab/cleanlab/pull/614/commits/31d86f4001578866ba1f2d9d1142d7ba8dcefccf#diff-e126cf653842496e91462249d07fa417cca997b74bc9924d4c80d9c715c11ea0R633

On the Serializer side:
https://github.com/cleanlab/cleanlab/pull/614/commits/31d86f4001578866ba1f2d9d1142d7ba8dcefccf#diff-28f21760451560d7c188f793deda7d3d828ead62dd02d672b47ee92540c27229R70